code/modules/mob/living/simple_animal/grue_egg.dm

```diff
@@ -35,6 +35,11 @@
 	var/last_ping_time = 0
 	var/ping_cooldown = 50
 
+	var/accum_light_expos_mult= 1 //used to scale light damage the longer the grue egg is exposed to light
+	var/list/accum_light_expos_gain_dark_dim_light=list(-3,-1,1) //light damage rate increases the longer the grue egg is exposed to light, but this effect dissipates after going back into darkness
```

Comment:
This var and others are duplicated between `grue_egg` and `grue`.
Have you considered using inheritance to avoid repeating yourself?

Reply:
One issue is that grue eggs are  `mob/living/simple_animal` while grues are `mob/living/simple_animal/hostile`, so I'm not sure what the ideal solution is.

Reply:
True. You could move these vars and procs to a datum and use it in both places.

Reply:
I think I got it set up as a datum.